File: test/fixtures/general/src/main/java/p1/ObjectCallReference.java
package p1;

public class ObjectCallReference {
    public void callMethod() {
        var client = new ObjectCallHelper();
        client.methodSelf();
        client.method();
    }

    public void callMethodChain() {
        var client = new ObjectCallHelper();
        client.methodSelf()
                .method();
    }
}

File: test/fixtures/general/src/main/kotlin/pkt1/ObjectCallReference.kt
package pkt1

class ObjectCallReference {
    fun callMethod() {
        val client = ObjectCallHelper()
        client.methodSelf()
        client.method()
    }

    fun callMethodChain() {
        val client = ObjectCallHelper()
        client.methodSelf()
            .method()
    }

    fun callMethodChainWithNullableAndHigherOrderFunctions() {
        val client = ObjectCallHelper()
        client.methodSelfWithNullable()
            ?.methodWithHigherOrderFunctions { }
    }
}